joblib/parallel.py

```diff
@@ -385,7 +385,9 @@ def effective_n_jobs(n_jobs=-1):
     .. versionadded:: 0.10
 
     """
-    backend, _ = get_active_backend()
+    backend, backend_n_jobs = get_active_backend()
+    if n_jobs is None:
+        return backend_n_jobs
```

Comment:
You should only return `backend_n_jobs` if it is set explicitly. It can be set to known to let the backend decide what is the correct default number of jobs (for instance 1 for loky but -1 for dask I believe).

Reply:
I am not sure about this. I think we should return `backend.effective_n_jobs(n_jobs=backend_n_jobs)` instead as suggested in the comment on the line below.

Reply:
You are right, this would be the best solution.